src/adtl/autoparser/mapping/long_mapper.py

```
@@ -0,0 +1,233 @@
+        """
+        if self.config.long_tables is None:
+            raise ValueError(
+                "No long tables defined in config file. Please set 'long_tables' in the config file."
+            )
+
+        if self.schema_fields[self.variable_col].get("enum") is None:
+            raise ValueError(
+                f"Variable column '{self.variable_col}' in schema does not have an enum set. "
+                "Please set 'enum' in the schema for this field."
+            )
+
+    def set_common_fields(self, common_fields: dict[str, str]):
+        """
+        Function to assign fields to the common fields of the long table - i.e. fields
+        which should be filled by the same text or source field in every row of the long
+        table.
+        """
+        if self.common_cols != list(common_fields.keys()):
+            raise ValueError(
+                f"Common columns {self.common_cols} set in the config file do not"
+                f" match provided common fields {common_fields.keys()}"
+            )
+
+        self._common_fields = common_fields
+        self._common_cols = list(common_fields.keys())
+
+    def match_fields_to_schema(self) -> pd.DataFrame:
+        """
+        Use the LLM to match the target (schema) fields to the descriptions of the
+        source data fields from the data dictionary.
+        """
+
+        data_format = self._create_data_structure()
+
+        if self.common_cols:
+            self.uncommon_data_dict = self.data_dictionary[
+                ~self.data_dictionary.source_field.isin(self.common_cols)
+            ].drop(columns=["source_type"])
+        else:
+            self.uncommon_data_dict = self.data_dictionary.drop(columns=["source_type"])
+
+        source_descriptions = self.uncommon_data_dict.source_description
+
+        mappings = self.model.map_long_table(
+            data_format,
+            source_descriptions.tolist(),
+            self.schema_fields[self.variable_col].get("enum", []),
```

if enum is empty list, this should raise an error?

---

ok, no need for an extra check here then